From 9b279a137e0ccaae5318e204caf8c3eda232c1a5 Mon Sep 17 00:00:00 2001
From: Bing Han <bing.han@intel.com>
Date: Wed, 6 Nov 2019 15:59:10 +0800
Subject: [PATCH 02/13] hypervisor: config: modify industry config

---
 hypervisor/scenarios/industry/vm_configurations.c | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/hypervisor/scenarios/industry/vm_configurations.c b/hypervisor/scenarios/industry/vm_configurations.c
index 188ee579b..d1da22aac 100644
--- a/hypervisor/scenarios/industry/vm_configurations.c
+++ b/hypervisor/scenarios/industry/vm_configurations.c
@@ -51,8 +51,7 @@ struct acrn_vm_config vm_configs[CONFIG_MAX_VM_NUM] = {
 		.vcpu_affinity = VM1_CONFIG_VCPU_AFFINITY,
 		.vuart[0] = {
 			.type = VUART_LEGACY_PIO,
-			.addr.port_base = COM1_BASE,
-			.irq = COM1_IRQ,
+			.addr.port_base = INVALID_COM_BASE,
 		},
 		.vuart[1] = {
 			.type = VUART_LEGACY_PIO,
@@ -72,15 +71,11 @@ struct acrn_vm_config vm_configs[CONFIG_MAX_VM_NUM] = {
 		.vcpu_affinity = VM2_CONFIG_VCPU_AFFINITY,
 		.vuart[0] = {
 			.type = VUART_LEGACY_PIO,
-			.addr.port_base = COM1_BASE,
-			.irq = COM1_IRQ,
+			.addr.port_base = INVALID_COM_BASE,
 		},
 		.vuart[1] = {
 			.type = VUART_LEGACY_PIO,
-			.addr.port_base = COM2_BASE,
-			.irq = COM2_IRQ,
-			.t_vuart.vm_id = 0U,
-			.t_vuart.vuart_id = 1U,
+			.addr.port_base = INVALID_COM_BASE,
 		},
 	},
 };
-- 
2.17.1

