From 3f6d57f5bb48d9367bebb27f7b296f5c07445b31 Mon Sep 17 00:00:00 2001
From: Naveen Saini <naveen.kumar.saini@intel.com>
Date: Thu, 14 Jul 2022 18:05:21 +0800
Subject: [PATCH 2/3] Makefile: allow to pass compiler and linker flags

Upstream-Status: Inappropriate [oe-specific]

Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 devicemodel/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/devicemodel/Makefile b/devicemodel/Makefile
index 372bab663..dbc77d3ff 100644
--- a/devicemodel/Makefile
+++ b/devicemodel/Makefile
@@ -45,6 +45,7 @@ CFLAGS += -I$(SYSROOT)/usr/include/glib-2.0
 CFLAGS += -I$(SYSROOT)/usr/include/SDL2
 CFLAGS += -I$(SYSROOT)/usr/include/EGL
 CFLAGS += -I$(SYSROOT)/usr/include/GLES2
+CFLAGS += -I$(COPTS)
 
 CFLAGS += -DIASL_MIN_VER=\"$(IASL_MIN_VER)\"
 
@@ -77,6 +78,7 @@ LDFLAGS += -Wl,-z,noexecstack
 LDFLAGS += -Wl,-z,relro,-z,now
 LDFLAGS += -pie
 LDFLAGS += -L$(TOOLS_OUT)/services
+LDFLAGS += -L$(COPTS)
 
 LIBS = -lrt
 LIBS += -lpthread
-- 
2.37.3

