From d44f7502ff0dca7ebe84cad06da469eb60865000 Mon Sep 17 00:00:00 2001
From: Naveen Saini <naveen.kumar.saini@intel.com>
Date: Tue, 9 Mar 2021 09:51:29 +0800
Subject: [PATCH] Makefile: allow to pass compiler and linker flags

Upstream-Status: Inappropriate [oe-specific]

Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 devicemodel/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/devicemodel/Makefile b/devicemodel/Makefile
index cea0b15b..95dd1b94 100644
--- a/devicemodel/Makefile
+++ b/devicemodel/Makefile
@@ -41,6 +41,7 @@ CFLAGS += -I$(BASEDIR)/include
 CFLAGS += -I$(BASEDIR)/include/public
 CFLAGS += -I$(DM_OBJDIR)/include
 CFLAGS += -I$(TOOLS_OUT)/services
+CFLAGS += -I$(COPTS)
 
 ifneq (, $(DM_ASL_COMPILER))
 CFLAGS += -DASL_COMPILER=\"$(DM_ASL_COMPILER)\"
@@ -75,6 +76,7 @@ LDFLAGS += -Wl,-z,noexecstack
 LDFLAGS += -Wl,-z,relro,-z,now
 LDFLAGS += -pie
 LDFLAGS += -L$(TOOLS_OUT)/services
+LDFLAGS += -L$(COPTS)
 
 LIBS = -lrt
 LIBS += -lpthread
-- 
2.17.1

